modules/gnome/nixos.nix

```diff
@@ -2,38 +2,46 @@
   lib,
   pkgs,
   config,
+  mkTarget,
   ...
 }:
+mkTarget {
+  name = "gnome";
+  humanName = "GNOME and GDM";
+  autoEnable =
+    config.services.xserver.desktopManager.gnome.enable
+    || config.services.xserver.displayManager.gdm.enable;
 
-let
-  theme = pkgs.callPackage ./theme.nix {
-    inherit (config.lib.stylix) colors;
-    inherit (config.stylix) inputs;
-  };
-
-in
-{
-  options.stylix.targets.gnome.enable =
-    config.lib.stylix.mkEnableTarget "GNOME and GDM" true;
-
-  config =
-    lib.mkIf
-      (
-        config.stylix.enable
-        && config.stylix.targets.gnome.enable
-        && (
-          config.services.xserver.desktopManager.gnome.enable
-          || config.services.xserver.displayManager.gdm.enable
-        )
-      )
+  configElements = [
+    {
+      # As Stylix is controlling the wallpaper, there is no need for this
+      # pack of default wallpapers to be installed.
+      # If you want to use one, you can set stylix.image to something like
+      # "${pkgs.gnome-backgrounds}/path/to/your/preferred/background"
+      # which will then download the pack regardless of its exclusion below.
+      environment.gnome.excludePackages = [ pkgs.gnome-backgrounds ];
+    }
+    (
+      { cursor }:
+      {
+        environment.systemPackages = [ cursor.package ];
+        programs.dconf.profiles.gdm.databases = [
+          {
+            lockAll = true;
+            settings."org/gnome/desktop/interface" = {
+              cursor-theme = config.stylix.cursor.name;
+              cursor-size = lib.gvariant.mkInt32 config.stylix.cursor.size;
+            };
+          }
+        ];
+      }
+    )
+    (
+      { colors, inputs }:
+      let
+        theme = pkgs.callPackage ./theme.nix { inherit colors inputs; };
+      in
       {
-        # As Stylix is controlling the wallpaper, there is no need for this
-        # pack of default wallpapers to be installed.
-        # If you want to use one, you can set stylix.image to something like
-        # "${pkgs.gnome-backgrounds}/path/to/your/preferred/background"
-        # which will then download the pack regardless of its exclusion below.
-        environment.gnome.excludePackages = [ pkgs.gnome-backgrounds ];
-
         nixpkgs.overlays = [
```

Comment:
> > I don't see any obvious inf-rec here, and CI seems happy atm?
> 
> see https://github.com/nix-community/stylix/actions/runs/15172177179/job/42664658225#step:5:18487

Seems related to this overlay.

If `pkgs` is being used as part of the condition for whether or not to enable this `configElement`, that'd make sense.

This PR makes the overlay conditional on `colors` and `inputs` being non-null. What guarded the overlay before?

Reply:
Previously it was just guarded by

```nix
(
  config.stylix.enable
  && config.stylix.targets.gnome.enable
  && (
    config.services.xserver.desktopManager.gnome.enable
    || config.services.xserver.displayManager.gdm.enable
  )
)
```

Reply:
> Infinite recursion only happens when using a generated scheme, which makes sense since we need to evaluate `pkgs` to produce that scheme.

Ah yes, I believe when I initially reviewed this I was unaware you were using IFD to import your generated schema.

It does make sense that it'd be infinite recursion in this case; possibly infinite rebuilds of the schema too...